fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/api/DatatablesApiResource.java

```diff
@@ -225,6 +228,18 @@ public String queryValues(@PathParam("datatable") @Parameter(description = "data
         return this.toApiJsonSerializer.serializePretty(ApiParameterHelper.prettyPrint(uriInfo.getQueryParameters()), result);
     }
 
+    @POST
+    @Path("{datatable}/query")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Query Data Table values", description = "Query values from a registered data table.")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = List.class))) })
+    public String advancedQuery(@PathParam("datatable") @Parameter(description = "datatable") final String datatable,
+            PagedRequest<AdvancedQueryData> queryRequest, @Context final UriInfo uriInfo) {
+        final Page<JsonObject> result = this.readWriteNonCoreDataService.queryDataTableAdvanced(datatable, queryRequest);
+        return this.toApiJsonSerializer.serializePretty(ApiParameterHelper.prettyPrint(uriInfo.getQueryParameters()), result);
```

Comment:
Since this is a new API, you should simply return a Page<....> and it'll be serialized automatically. That way you can drop all the openAPI annotations as well.

Reply:
I answered this request once, please let me know the rules, because so far I learned that only v2 APIs can use object response by decision, and this one was added as v1 API.

Reply:
The rule is
- for V2 it is mandatory to be object response
- for V1 **new** API it is okay to be object response